torch/nn/parallel/distributed.py

```diff
@@ -606,9 +606,10 @@ def produces_sparse_gradient(module):
             self.find_unused_parameters,
             self.gradient_as_bucket_view)
 
+        self.logger = dist.Logger(self.reducer)
```

Comment:
nit: if `dist.Logger` is not meant to be publicly accessed, let's use it as `dist._Logger`.

Reply:
we want it to be public, e.g. users can use dist.logger to get logging data